libraries/search/lib/objective/branch/BranchObjectiveFunction.ts

```diff
@@ -60,47 +64,14 @@ export class BranchObjectiveFunction<
    * @param encoding
    * @returns
    */
-  // eslint-disable-next-line sonarjs/cognitive-complexity
-  calculateDistance(encoding: T): number {
-    const executionResult = encoding.getExecutionResult();
-
-    if (
-      executionResult === undefined ||
-      executionResult.getTraces().length === 0
-    ) {
-      return Number.MAX_VALUE;
-    }
-
-    // check if the branch is covered
-    if (executionResult.coversId(this._id)) {
-      return 0;
-    } else if (this.shallow) {
-      return Number.MAX_VALUE;
-    } else {
-      return this._calculateControlFlowDistance(executionResult);
-    }
-  }
-
   // eslint-disable-next-line sonarjs/cognitive-complexity
   protected _calculateControlFlowDistance(
+    id: string,
```

Comment:
You don't need this id right? it is an internal property

Reply:
Its such that the pathobjectivefunction can give a different id as argument. The path objective needs to check multiple ids instead of just one.

Reply:
Maybe you could move it to the common base class then